### PR TITLE
[combat-trainer] Don't manipulate constructs

### PR DESCRIPTION
**Pull request by Ry7n**
Ensures if you run `;combat-trainer construct` you won't manipulate mobs, and adds a catch if you accidentally do (e.g. in mixed groups).

### CODE REVIEW

File: combat-trainer.lic [hunk 1]
```diff
@@ -1434,11 +1434,14 @@ class ManipulateProcess
     @threshold = settings.manipulate_threshold
     echo("  @threshold: #{@threshold}") if $debug_mode_ct
 
+    @construct_mode = settings.construct(false)
```
Comment (rpherbig) [inline combat-trainer.lic]: What would you think about moving this into a predicate in `GameState`? There's a lot of duplication with `AttackProcess` here

File: combat-trainer.lic [hunk 2]
```diff
@@ -1456,14 +1459,19 @@ class ManipulateProcess
     npc_occurrences = Hash.new(0)
 
     @filtered_npcs.each do |npc|
+      next if construct?(npc)
```
Comment (rpherbig) [inline combat-trainer.lic]: Does this need to be `game_state.construct?(npc)`?

File: combat-trainer.lic [hunk 3]
```diff
@@ -1456,14 +1459,19 @@ class ManipulateProcess
...
 
       index = npc_occurrences[npc]
       ordinal_string = $ORDINALS[index]
 
-      bput("manipulate friendship #{ordinal_string} #{npc}", 'You\'re already manipulating', 'beyond your ken', 'You attempt to empathically manipulate', 'You strain')
-      manipulate_count += 1
-      npc_occurrences[npc] += 1
+      case bput("manipulate friendship #{ordinal_string} #{npc}", 'You\'re already manipulating', 'beyond your ken', 'You attempt to empathically manipulate', 'You strain', 'no life force')
+      when 'no life force' then
+        construct(npc)
```
Comment (rpherbig) [inline combat-trainer.lic]: Does this need to be `game_state.construct(npc)`?

File: combat-trainer.lic (hunk 1 above)
Comment (Ry7n) [reply combat-trainer.lic]: Sure thing.

**Pull request merged**